### PR TITLE
Int 520 removed extended sdk from intellij

### PR DESCRIPTION
**Pull request by aliaandersen**
This adds the necessary headers to the makeConnection method to handle the incoming telemetry values. 

This adds the necessary files to remove the extendedSDK from intellij. 

Bumps java version to 1.8 and updates README accordingly.

Bumps major version for the java change and removes depreciated constructors.

### CODE REVIEW

File: src/main/java/com/contrastsecurity/sdk/ContrastSDK.java [hunk 1]
```diff
@@ -151,7 +160,7 @@ public ContrastSDK(String user, String serviceKey, String apiKey, String restApi
         ContrastSDKUtils.validateUrl(this.restApiURL);
         this.restApiURL = ContrastSDKUtils.ensureApi(this.restApiURL);
         this.urlBuilder = UrlBuilder.getInstance();
-        this.gson = new GsonBuilder()
+        this.gson = new GsonBuilder().excludeFieldsWithoutExposeAnnotation()
```
Comment (bryanateoan) [inline src/main/java/com/contrastsecurity/sdk/ContrastSDK.java:163]: I think this breaks serialized objects that were never tagged with expose like JobOutcomePolicyListResponse

Comment (bryanateoan) [reply src/main/java/com/contrastsecurity/sdk/ContrastSDK.java:163]: Maybe we should us transient instead of expose annotation? what do you think?

**Review by bryanateoan — CHANGES_REQUESTED** [review]
Requested some changes and discussion questions.

File: .java-version [hunk 2]
```diff
@@ -1 +0,0 @@
-openjdk64-11.0.2
```
Comment (bryanateoan) [inline .java-version:1]: Is this expected?

File: src/main/java/com/contrastsecurity/models/Tags.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package com.contrastsecurity.models;
+
+import com.google.gson.annotations.Expose;
```
Comment (bryanateoan) [inline src/main/java/com/contrastsecurity/models/Tags.java:3]: Lets remove if no longer needed.

File: pom.xml [hunk 4]
```diff
@@ -107,14 +107,6 @@
             </testResource>
         </testResources>
         <plugins>
-            <plugin>
-                <artifactId>maven-compiler-plugin</artifactId>
```
Comment (bryanateoan) [inline pom.xml:111]: Please update the README and make note in the release notes.

**Review by bryanateoan — CHANGES_REQUESTED** [review]

File: .java-version (hunk 2 above)
Comment (aliaandersen) [reply .java-version:1]: Nope, I'll add to gitignore.

File: pom.xml [hunk 5]
```diff
@@ -10,7 +10,7 @@
 
     <groupId>com.contrastsecurity</groupId>
     <artifactId>contrast-sdk-java</artifactId>
-    <version>2.23-SNAPSHOT</version>
+    <version>3.0</version>
```
Comment (bryanateoan) [inline pom.xml:13]: Keep it 3.0-SNAPSHOT. The release process will release 3.0 and move the development version to 3.1-SNAPSHOT.

File: src/main/java/com/contrastsecurity/http/UrlBuilder.java [hunk 6]
```diff
@@ -102,9 +102,41 @@ public String getTracesByApplicationUrl(String organizationId, String appId, Tra
...
+        return String.format("/ng/%s/traces/%s/events/summary", organizationId, traceId);
+    }
+
+    public String getEventDetails(String organizationId, String traceId, String eventId){
+        return String.format("/ng/%s/traces/%s/events/%s/details", organizationId, traceId, eventId);
+    }
+
     public String getNotesByApplicationUrl(String organizationId, String appId, String traceId, TraceFilterForm form) throws UnsupportedEncodingException {
         String formString = form == null ? "" : form.toQuery();
-        return String.format("/ng/%s/applications/%s/traces/%s/notes?expand=skip_links", organizationId, appId, traceId, formString);
```
Comment (bryanateoan) [inline src/main/java/com/contrastsecurity/http/UrlBuilder.java:107]: Why did you remove the expand=skip_links?

Comment (bryanateoan) [reply src/main/java/com/contrastsecurity/http/UrlBuilder.java:107]: is that no longer needed?

**Review by bryanateoan — CHANGES_REQUESTED** [review]
* Change the version to 3.0-SNAPSHOT
* Question about getNotesByApplicationURL

Rest looks good.

File: pom.xml (hunk 5 above)
Comment (aliaandersen) [reply pom.xml]: Shoot, got to excited moving to v3

File: src/main/java/com/contrastsecurity/http/UrlBuilder.java (hunk 6 above)
Comment (aliaandersen) [reply src/main/java/com/contrastsecurity/http/UrlBuilder.java:107]: Yeah, it wasn't necessary for this endpoint

Comment (aliaandersen) [reply src/main/java/com/contrastsecurity/http/UrlBuilder.java:107]: This endpoint, as far as I know, doesn't return links, so it shouldn't need the expand param

**Review by bryanateoan — APPROVED** [review]

File: src/main/java/com/contrastsecurity/sdk/ContrastSDK.java [hunk 7]
```diff
@@ -124,74 +133,19 @@ public ContrastSDK build() {
         }
     }
 
-    /**
-     * Use ContrastSDK.Builder
-     */
-    @Deprecated
     public ContrastSDK() {
```
Comment (bryanateoan) [inline src/main/java/com/contrastsecurity/sdk/ContrastSDK.java:136]: lets make this protected with a visible for testing comment.

File: src/main/java/com/contrastsecurity/sdk/ContrastSDK.java [hunk 8]
```diff
@@ -64,11 +72,12 @@
     private String apiKey;
     private String serviceKey;
     private String user;
-    private String restApiURL;
+    @Getter
+    private static String restApiURL;
```
Comment (bryanateoan) [inline src/main/java/com/contrastsecurity/sdk/ContrastSDK.java:76]: Why is this static?

**Review by bryanateoan — CHANGES_REQUESTED** [review]
just a couple of items, rest looks good.

**Review by bryanateoan — APPROVED** [review]
